### PR TITLE
status: 2023q2: epochification of netgraph

### PR DESCRIPTION
**Pull request by zinh88**
gsoc 2023

### CODE REVIEW

File: website/content/en/status/report-2023-04-2023-06/epoch-netgraph.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+We want to regain the same smooth flow for data which existed when concurrent cpus were not a thing. That is, data should simply never wait every time there is a restructuring event. At the same time we also obviously do not want to give the kernel reasons to panic. 
+
+FreeBSD has its own set of concurrency-safe data structures and mechanisms. One of these mechanisms is Epoch. Epoch-based reclamation involves waiting for existing read-side critical sections to finish before the data structures need to be modified or reclaimed.
+
+Because the base system is being modified, this is also going to affect the design choices made before, such as queuing on messages, reference counting. 
+
+This project involves a lot of testing. For now, some topology protection locks have been removed, and only simple graphs have been tested (with FreeBSD running on a VM). The real tests should be run on hardware with at least 4 CPU cores, I'll do that when I get my hands on one.
```
Comment (lsalvadore) [inline website/content/en/status/report-2023-04-2023-06/epoch-netgraph.adoc:19]:
```suggestion
This project involves a lot of testing. For now, some topology protection locks have been removed, and only simple graphs have been tested (with FreeBSD running on a VM). The real tests should be run on hardware with at least 4 CPU cores, I will do that when I get my hands on one.
```

Please avoid contractions.

**Review by lsalvadore — CHANGES_REQUESTED** [review]
Thank you very much for your report.

There are just two things to fix and then we can proceed to merge the report. Please also note that we usually require to write one sentence per line in adoc files, but this can be fixed at merge time to speed up things, you do not need to do it now.

I guess you do not have commit access and you need me to do the merge, am I right?

Thanks.

File: website/content/en/status/report-2023-04-2023-06/epoch-netgraph.adoc (hunk 1 above)
Comment (zinh88) [reply website/content/en/status/report-2023-04-2023-06/epoch-netgraph.adoc:19]: I am so sorry for any mistakes. This whole process is very new to me and I was informed that I had to do this. Change whatever is necessary or just tell me what to do. 
No, I do not have commit access. 

Best,
Zain

Comment (lsalvadore) [reply website/content/en/status/report-2023-04-2023-06/epoch-netgraph.adoc:19]: Do not worry, you have written a very good report and the few things to fix are only minor changes. It happens that more experienced developers in the FreeBSD community need to go through longer reviews with more changes needed.
Keep up the good work.

I will take care of the changes needed here and merge the report as soon as possible.

Thanks.

**Comment by lsalvadore** [comment]
Merged, thanks.

**Pull request closed without merging**